### PR TITLE
[Logs] Updated Max Upload Parameters and created a page for Log Output Options

### PR DESCRIPTION
**Pull request by angelampcosta**

### CODE REVIEW

File: content/logs/get-started/api-configuration.md [hunk 1]
```diff
@@ -201,13 +168,16 @@ Value can range from `0.001` to `1.0` (inclusive). `sample=0.1` means `return 10
...
 
 1.  **max_upload_bytes** (optional): The maximum uncompressed file size of a batch of logs. This must be at least 5 MB. Note that you cannot set a minimum file size; this means that log files may be much smaller than this batch size.
 2.  **max_upload_records** (optional): The maximum number of log lines per batch. This must be at least 1000 lines or more. Note that you cannot specify a minimum number of log lines per batch; this means that log files may contain many fewer lines than this.
+3.  **max_upload_interval_seconds** (optional): The maximum interval in seconds for log batches. This setting must be between 30 and 300 seconds. Note that you cannot specify a minimum interval for log batches; this means that log files may be sent in shorter intervals than this.
 
 {{<Aside type="note" header="Note">}}
-Parameters **max_upload_bytes** and **max_upload_records** are not configurable for Edge Log Delivery.
+- There are no maximum value limits for these parameters, however, setting them too high may result in data loss.
```
Comment (soheiokamoto) [inline content/logs/get-started/api-configuration.md]: Each of them actually has maximum value limits.
I.e., current allowed ranges are:
 - max_upload_bytes: 5MB to 1GB
 - max_upload_records: 1K to 1M
 - max_upload_interval_seconds: 30 seconds to 5 minutes (as describe above)

File: content/logs/get-started/log-output-options.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+- **record_template**: string to use as template for each record instead of the default comma-separated list. All fields used in the template must be present in **field_names** as well, otherwise they will end up as `null`. Format as a Go text/template without any standard functions (like conditionals, loops, sub-templates, etc.). The template can only consist of these three types of tokens:
+    - Action: this is either a `{{ .Field }}` or a `{{ "constant text" }}`.
+    - Text: this is just constant text in-between the `{{ actions }}`.
+    - Comment: the `{{/* comments */}}` are silently dropped.
+- **record_delimiter**: string to be inserted in-between the records as separator.
+
+- **field_delimiter**: string to join fields. Will be ignored when **record_template** is set.
+
+- **timestamp_format**: string to specify format for timestamps, such as `unixnano`, `unix`, or `rfc3339`. Default `unixnano`.
+- **sample_rate**: floating number to specify sampling rate (default 1.0: no sampling). Sampling is applied on top of andregardless of the current sample_interval of the data.
```
Comment (soheiokamoto) [inline content/logs/get-started/log-output-options.md]: I think this should be something like:
```suggestion
- **sample_rate**: floating number to specify sampling rate (default 1.0: no sampling). Sampling is applied on top of filtering, and regardless of the current sample_interval of the data.
```

File: content/logs/get-started/log-output-options.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
+---
```
Comment (soheiokamoto) [inline content/logs/get-started/log-output-options.md]: Shouldn't this page go under `Reference`, just like `Filter` and `Custom fields`?

File: content/logs/get-started/api-configuration.md (hunk 1 above)
Comment (tlozoot) [reply content/logs/get-started/api-configuration.md]: Can we address this before merge?

File: content/logs/get-started/log-output-options.md (hunk 3 above)
Comment (tlozoot) [reply content/logs/get-started/log-output-options.md]: I agree with this

File: content/logs/get-started/log-output-options.md [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+weight: 43
+---
+
+# Log Output Options
+
+Jobs in Logpush now have a new key **output_options** which replaces **logpull_options** and allows more flexible formatting.
+
+Edge Logstream jobs do not support this yet.
+
+## Why replace logpull_options
```
Comment (tlozoot) [inline content/logs/get-started/log-output-options.md]: "Why" is a bit odd in section header here -- we don't really explain "why" and not sure we need to put that in dev docs (maybe a blog post?)

Maybe just "Replacing logpull_options"?

File: content/logs/get-started/api-configuration.md [hunk 5]
```diff
@@ -156,40 +156,7 @@ curl -s -X POST 'https://api.cloudflare.com/client/v4/zones/<ZONE_ID>/logpush/jo
 
 ## Options
 
-Logpush repeatedly delivers logs on your behalf and uploads them to your destination.
-
-The options that you can customize are:
```
Comment (tlozoot) [inline content/logs/get-started/api-configuration.md:161]: Can we keep this somewhere in the docs? I'm sure there are people who are still using logpull_options and will need time to migrate, and they'll want to see what the old options were.

**Review by tlozoot — CHANGES_REQUESTED** [review]
This is great -- just a few minor comments before merge

File: content/logs/get-started/log-output-options.md (hunk 3 above)
Comment (angelampcosta) [reply content/logs/get-started/log-output-options.md]: Makes sense. I'll move it.

File: content/logs/get-started/api-configuration.md (hunk 1 above)
Comment (angelampcosta) [reply content/logs/get-started/api-configuration.md]: I'll add the limits.

File: content/logs/get-started/api-configuration.md (hunk 5 above)
Comment (angelampcosta) [reply content/logs/get-started/api-configuration.md:161]: I think we can keep the options were they are, after saying that logpull_options is substituted by output_options. When we think it's no longer needed, we can remove them from the docs.

File: content/logs/reference/log-output-options.md [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+
+## Output types
+
+By default Logpush outputs each record as a single line of JSON (aka ndjson).
+
+With **output_options** you can switch to CSV or single JSON object, further customize prefixes, suffixes, delimiters, or provide your own record template (in a stripped-down version of Go [text/template](https://pkg.go.dev/text/template) syntax).
+
+The **output_options** object has the following settings:
+
+- **field_names**: array of strings.
```
Comment (victor-perov) [inline content/logs/reference/log-output-options.md:53]: nit: if we group options here by separating them with empty space then `field_names` should be accompanied with `field_delimeter` and separated from `output_type` (which in its turn should be grouped with `CVE-2021-44228`).

File: content/logs/reference/log-output-options.md [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+```
+
+</div>
+</details>
+
+<details>
+<summary>Example output</summary>
+<div>
+
+```json
```
Comment (victor-perov) [inline content/logs/reference/log-output-options.md]: looks like a CSV to me

File: content/logs/reference/log-output-options.md [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
```
Comment (victor-perov) [inline content/logs/reference/log-output-options.md]:
```suggestion
```csv
```

File: content/logs/reference/log-output-options.md [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
```
Comment (victor-perov) [inline content/logs/reference/log-output-options.md]:
```suggestion
```csv
```

File: content/logs/reference/log-output-options.md (hunk 7 above)
Comment (victor-perov) [inline content/logs/reference/log-output-options.md]:
```suggestion
```csv
```

File: content/logs/reference/log-output-options.md [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
+  {"info":{"ClientIP":"89.163.242.206","EdgeStartTimestamp":1506702504433000200,"RayID":"3a6050bcbe121a87"}},
+  {"info":{"ClientIP":"89.163.242.207","EdgeStartTimestamp":1506702504433000300,"RayID":"3a6050bcbe121a88"}},
+  {"info":{"ClientIP":"89.163.242.208","EdgeStartTimestamp":1506702504433000400,"RayID":"3a6050bcbe121a89"}}
+]}
+```
+
+</div>
+</details>
+
+- ndjson with different field names:
```
Comment (victor-perov) [inline content/logs/reference/log-output-options.md]: nit: move this example to the `ndjson` section above

**Review by victor-perov — CHANGES_REQUESTED** [review]
Thanks for working on it! I think we should fix json/csv in some example headers to make the syntax clearer.

File: content/logs/reference/log-output-options.md (hunk 6 above)
Comment (angelampcosta) [reply content/logs/reference/log-output-options.md:53]: I removed the extra lines because they weren't displaying.

**Comment by jsells1** [comment]
👍

**Review by victor-perov — APPROVED** [review]

File: content/logs/reference/log-output-options.md [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+
+Jobs in Logpush now have a new key **output_options** which replaces **logpull_options** and allows more flexible formatting.
+
+Edge Logstream jobs do not support this yet.
+
+## Replacing logpull_options
+
+Previously, Logpush jobs could be customized by specifying the list of fields, sampling rate, and timestamp format in **logpull_options** as [URL-encoded parameters](/logs/get-started/api-configuration/#options). For example:
+
+```JSON
```
Comment (soheiokamoto) [inline content/logs/reference/log-output-options.md]: I don't know if it matters, but shouldn't these be lowercase usually?
```suggestion
```json
```

**Review by tlozoot — APPROVED** [review]

**Pull request merged**